### PR TITLE
Make the go program aware of its own version

### PR DESCRIPTION
**Pull request by shubhitms**
This simply makes it so that the go program can read the version number from a file and embed it 
into the source code. 

as a follow up i will write a release script that updates the version.txt file

### CODE REVIEW

File: internal/version/version.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	_ "embed"
+	"fmt"
+	"strings"
+)
+
+var (
+	//go:embed version.txt
+	version    string
+	Version    = readVersion()
+	SDKVersion = fmt.Sprintf("go-%s", Version)
```
Comment (DavidSGK) [inline internal/version/version.go:27]: Maybe OOTL, what's the motivation for having 2 different types of versions (where the only difference is a `go-` prefix)?

Comment (shubhitms) [reply internal/version/version.go]: one is for unit tests (to make sure its semver), and the other is prefixed with `go-` for sending back to the backend. I just changed to only make the latter public

**Review by DavidSGK — APPROVED** [review]
The changes here look good, but I wonder if there's a way we could use git/our build system to include version info in the SDK libraries in a way that doesn't depend on an embedded text file.

In the [Go docs](https://tip.golang.org/doc/go1.18#go-version) I see that building with the go command includes VCS info by default but this is for binaries and doesn't mention libraries - is this something you've seen?

**Comment by shubhitms** [comment]
> The changes here look good, but I wonder if there's a way we could use git/our build system to include version info in the SDK libraries in a way that doesn't depend on an embedded text file.
> 
> In the [Go docs](https://tip.golang.org/doc/go1.18#go-version) I see that building with the go command includes VCS info by default but this is for binaries and doesn't mention libraries - is this something you've seen?

It wouldn't work for the SDK because we don't control the build step for our customers. If our customers do rely on `go version` in their own build step, it will embed the VCS info of their parent application, not our SDK. This will apply to every SDK we release as well, not just Go.

**Comment by shubhitms** [comment]
> The changes here look good, but I wonder if there's a way we could use git/our build system to include version info in the SDK libraries in a way that doesn't depend on an embedded text file.
> 
> In the [Go docs](https://tip.golang.org/doc/go1.18#go-version) I see that building with the go command includes VCS info by default but this is for binaries and doesn't mention libraries - is this something you've seen?

actually, it looks like the `go version` functionality includes dependency versions. also, go 1.17 and before are now [EOL](https://endoflife.date/go), so i think its fine to assume clients will be running >=1.18. i've updated to remove the txt file and rely on `runtime/debug` instead.

**Pull request merged**